code/modules/mob/living/living_defense.dm

```diff
@@ -306,6 +307,7 @@
 	return FALSE
 
 /mob/living/attack_alien(mob/living/carbon/alien/humanoid/user, list/modifiers)
+	SEND_SIGNAL(src, COMSIG_MOB_ATTACK_ALIEN, user)
 	if(LAZYACCESS(modifiers, RIGHT_CLICK))
```

Comment:
Should add should call parent here

Reply:
I agree, that way `COMSIG_MOB_ATTACK_ALIEN` isn't necessary and you can just use one signal.

Reply:
Nope, it'll otherwise call attack_paw too:
https://github.com/tgstation/tgstation/blob/f98ffcb19a29e34a8bf294b3e66dd61f6874b673/code/_onclick/other_mobs.dm#L154-L155

There are also severals other movables that interact differently with xenos, A rather not modest attack_alien() refactor would have to be done first.